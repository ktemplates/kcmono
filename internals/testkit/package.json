{
  "name": "@internal/testkit",
  "private": true,
  "version": "0.1.0",
  "main": "lib/index.js",
  "typings": "lib/index.d.ts",
  "scripts": {
    "build": "icompiler-tsc",
    "start": "irunner",
    "clean": "icleaner"
  },
  "dependencies": {
    "react": "16.13.1",
    "react-dom": "16.13.1",
    "enzyme": "3.11.0",
    "enzyme-adapter-react-16": "1.15.2",
    "enzyme-to-json": "3.5.0"
  },
  "devDependencies": {
    "@internal/commandline": "0.1.0",
    "@types/enzyme": "3.10.5",
    "@types/enzyme-adapter-react-16": "1.0.6"
  }
}
