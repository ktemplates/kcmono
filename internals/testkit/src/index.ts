import Enzyme, { mount, render, shallow, toJson } from "./models/enzyme";

export { Enzyme, mount, render, shallow, toJson };
