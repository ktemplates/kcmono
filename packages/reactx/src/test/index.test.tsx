import React from "react";

import { shallow, toJson } from "@internal/testkit";
import { Hello } from "..";

describe("test", () => {
  test("create react component", () => {
    const component = shallow(<Hello compiler="typescript" framework="webpack" />);

    console.log(toJson(component));
    expect(toJson(component)).toMatchSnapshot();
  });
});
