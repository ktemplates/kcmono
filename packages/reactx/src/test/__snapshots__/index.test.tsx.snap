// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test create react component 1`] = `
<h1>
  Hello from 
  typescript
   and 
  webpack
  !
</h1>
`;
